### PR TITLE
Update CUDA version

### PR DESCRIPTION
**Pull request by bsewp045**
# Description

**What your PR adds/fixes/removes**

# Tests / Quality Checks

## Are there breaking changes?
Ask yourself the next question;
- [ ] Do we want to maintain the previous image from which we had to do breaking changes from?

If no, then carry on. If yes, there is a breaking change and we **want to maintain the previous image** do the following
- [ ] Create a new branch for the current version (ex v1) based off the current master/main branch
- [ ]  Increment the tag in the CI for pushes to master/main (v1 to v2)
- [ ] Change the CI that on pushes to the newly created "v1" branch (the name of the newly created branch we want to maintain is) it will push to the ACR. 
## Automated Testing/build and deployment
- [ ] Does the image pass CI successfully (build, pass vulnerability scan, and pass automated test suite)?
- [ ] If new features are added (new image, new binary, etc), have new automated tests been added to cover these?
- [ ] If new features are added that require in-cluster testing (e.g. a new feature that needs to interact with kubernetes), have you added the `auto-deploy` tag to the PR before pushing in order to build and push the image to ACR so you can test it in cluster as a custom image?

## JupyterLab extensions

- [ ] Are all extensions "enabled" (`jupyter labextension list` from inside the notebook)?

## VS Code tests

- [ ] Does VS Code open?
- [ ] Can you install extensions?

## Code review

- [ ] Have you added the `auto-deploy` tag to your PR before your most recent push to this repo?  This causes CI to build the image and push to our ACR, letting reviewers access the built image without having to create it themselves
- [ ] Have you chosen a reviewer, attached them as a reviewer to this PR, and messaged them with the SHA-pinned image name for the final image to test on the **dev cluster** (e.g. `k8scc01covidacrdev.azurecr.io/jupyterlab-cpu:746d058e2f37e004da5ca483d121bfb9e0545f2b`)?

### CODE REVIEW

File: .github/workflows/build_push.yaml [hunk 1]
```diff
@@ -82,7 +82,7 @@ jobs:
     - name: Set ENV variables for a PR containing the auto-deploy tag
       if: github.event_name == 'pull_request' && contains( github.event.pull_request.labels.*.name, 'auto-deploy')
       run: | 
-        echo "REGISTRY=k8scc01covidacrdev.azurecr.io" >> "$GITHUB_ENV"
+        echo "REGISTRY=k8scc01covidacr.azurecr.io" >> "$GITHUB_ENV"
```
Comment (Jose-Matsuda) [inline .github/workflows/build_push.yaml:85]: Change this back to `dev`

File: docker-bits/1_CUDA-11.6.Dockerfile [hunk 2]
```diff
@@ -1,27 +1,27 @@
...
+# Cuda stuff for v11.6.2
 
-## https://gitlab.com/nvidia/container-images/cuda/-/raw/master/dist/11.0/ubuntu18.04-x86_64/base/Dockerfile
+## https://gitlab.com/nvidia/container-images/cuda/-/blob/92d100e5ad724656d3b7315db8ce268ab7cb9d91/dist/11.6.2/ubuntu1804/base/Dockerfile
 
 ###########################
 ### Base
 ###########################
-# https://gitlab.com/nvidia/container-images/cuda/-/raw/master/dist/11.0/ubuntu18.04-x86_64/base/Dockerfile
+# https://gitlab.com/nvidia/container-images/cuda/-/blob/92d100e5ad724656d3b7315db8ce268ab7cb9d91/dist/11.6.2/ubuntu1804/base/Dockerfile
```
Comment (Jose-Matsuda) [inline docker-bits/1_CUDA-11.6.Dockerfile:8]: There any specific reason why we did not go with `raw` for this as well? As in https://gitlab.com/nvidia/container-images/cuda/-/raw/92d100e5ad724656d3b7315db8ce268ab7cb9d91/dist/11.6.2/ubuntu1804/base/Dockerfile

File: docker-bits/1_CUDA-11.6.Dockerfile [hunk 3]
```diff
@@ -34,41 +34,42 @@ ENV LD_LIBRARY_PATH /usr/local/nvidia/lib:/usr/local/nvidia/lib64
...
 ENV NVIDIA_VISIBLE_DEVICES all
 ENV NVIDIA_DRIVER_CAPABILITIES compute,utility
-ENV NVIDIA_REQUIRE_CUDA "cuda>=11.0 brand=tesla,driver>=418,driver<419 brand=tesla,driver>=440,driver<441 brand=tesla,driver>=450,driver<451"
+ENV NVIDIA_REQUIRE_CUDA "cuda>=11.6 brand=tesla,driver>=418,driver<419 brand=tesla,driver>=450,driver<451 brand=tesla,driver>=470,driver<471 brand=unknown,driver>=470,driver<471 brand=nvidia,driver>=470,driver<471 brand=nvidiartx,driver>=470,driver<471 brand=quadrortx,driver>=470,driver<471"
 
 # ###########################
 # ### Devel
 # ###########################
-# # https://gitlab.com/nvidia/container-images/cuda/-/raw/master/dist/11.0/ubuntu18.04-x86_64/devel/Dockerfile
+# # https://gitlab.com/nvidia/container-images/cuda/-/blob/92d100e5ad724656d3b7315db8ce268ab7cb9d91/dist/11.6.2/ubuntu1804/devel/Dockerfile
```
Comment (Jose-Matsuda) [inline docker-bits/1_CUDA-11.6.Dockerfile:42]: Same as above re: raw vs blob

File: docker-bits/1_CUDA-11.6.Dockerfile [hunk 4]
```diff
@@ -34,41 +34,42 @@ ENV LD_LIBRARY_PATH /usr/local/nvidia/lib:/usr/local/nvidia/lib64
...
 #
-# $(curl -s https://gitlab.com/nvidia/container-images/cuda/-/raw/master/dist/11.0/ubuntu18.04-x86_64/devel/Dockerfile)
+# $(curl -s https://gitlab.com/nvidia/container-images/cuda/-/blob/92d100e5ad724656d3b7315db8ce268ab7cb9d91/dist/11.6.2/ubuntu1804/devel/Dockerfile)
```
Comment (Jose-Matsuda) [inline docker-bits/1_CUDA-11.6.Dockerfile:44]: Same as above re: raw vs blob

File: docker-bits/1_CUDA-11.6.Dockerfile [hunk 5]
```diff
@@ -34,41 +34,42 @@ ENV LD_LIBRARY_PATH /usr/local/nvidia/lib:/usr/local/nvidia/lib64
...
 
 ###########################
 ### Runtime
 ###########################
-# https://gitlab.com/nvidia/container-images/cuda/-/raw/master/dist/11.0/ubuntu18.04-x86_64/runtime/Dockerfile
+# https://gitlab.com/nvidia/container-images/cuda/-/blob/92d100e5ad724656d3b7315db8ce268ab7cb9d91/dist/11.6.2/ubuntu1804/runtime/Dockerfile
```
Comment (Jose-Matsuda) [inline docker-bits/1_CUDA-11.6.Dockerfile:49]: Same as above re: raw vs blob

File: docker-bits/1_CUDA-11.6.Dockerfile [hunk 6]
```diff
@@ -34,41 +34,42 @@ ENV LD_LIBRARY_PATH /usr/local/nvidia/lib:/usr/local/nvidia/lib64
...
+    libcublas-11-6=11.9.2.110-1 \
+    libnccl2=$NCCL_VERSION-1+cuda11.6 \
     && apt-mark hold libnccl2 \
     && rm -rf /var/lib/apt/lists/*
 
 ###########################
 ### CudNN
 ###########################
-# https://gitlab.com/nvidia/container-images/cuda/-/raw/master/dist/11.0/ubuntu18.04-x86_64/runtime/cudnn8/Dockerfile
+# https://gitlab.com/nvidia/container-images/cuda/-/blob/92d100e5ad724656d3b7315db8ce268ab7cb9d91/dist/11.6.2/ubuntu1804/runtime/cudnn8/Dockerfile
```
Comment (Jose-Matsuda) [inline docker-bits/1_CUDA-11.6.Dockerfile:66]: Same as above re: raw vs blob

**Review by Jose-Matsuda — CHANGES_REQUESTED** [review]
I am guessing you did the same two tests here on prod as well correct? Where you started up a gpu machine and found that the gpu was accessible?

File: .github/workflows/build_push.yaml (hunk 1 above)
Comment (bsewp045) [reply .github/workflows/build_push.yaml:85]: Yeah, I tested them on prod.

File: docker-bits/1_CUDA-11.6.Dockerfile (hunk 3 above)
Comment (bsewp045) [reply docker-bits/1_CUDA-11.6.Dockerfile:42]: Same as above

File: docker-bits/1_CUDA-11.6.Dockerfile (hunk 4 above)
Comment (bsewp045) [reply docker-bits/1_CUDA-11.6.Dockerfile:44]: same as above

File: docker-bits/1_CUDA-11.6.Dockerfile (hunk 5 above)
Comment (bsewp045) [reply docker-bits/1_CUDA-11.6.Dockerfile:49]: same as above

File: docker-bits/1_CUDA-11.6.Dockerfile (hunk 6 above)
Comment (bsewp045) [reply docker-bits/1_CUDA-11.6.Dockerfile:66]: same as above

**Pull request merged**